backtesting/backtesting.py

```diff
@@ -405,18 +405,20 @@ def _replace(self, **kwargs):
             setattr(self, f'_{self.__class__.__qualname__}__{k}', v)
         return self
 
+    
     def __repr__(self):
-        return '<Order {}>'.format(', '.join(f'{param}={round(value, 5)}'
+        return '<Order {}>'.format(', '.join(f'{param}={value}'
                                              for param, value in (
-                                                 ('size', self.__size),
-                                                 ('limit', self.__limit_price),
-                                                 ('stop', self.__stop_price),
-                                                 ('sl', self.__sl_price),
-                                                 ('tp', self.__tp_price),
+                                                 ('size', round(self.__size,5)),
+                                                 ('limit', round(self.__limit_price,5) if self.__limit_price is not None else None),
+                                                 ('stop', round(self.__stop_price,5) if self.__stop_price is not None else None),
+                                                 ('sl', round(self.__sl_price,5) if self.__sl_price is not None else None),
+                                                 ('tp', round(self.__tp_price,0) if self.__tp_price is not None else None),
```

Comment:
I'd prefer this `round(value, 5) if value is not None else ''` be done once in the comprehension value expression (as before).

Also don't see a problem with `round(is_contingent, 5)` since the boolean rounds to either `1` or `0`.

Reply:
How would you prefer to deal with string tags then? Take them out of the comprehension and then append them separately?

Reply:
After some further investigation it could be:
```python
    def __repr__(self):
        return '<Order {}>'.format(', '.join(f'{param}={round(value, 5) if isinstance(value, SupportsRound) else value}'
                                             for param, value in (
                                                 ('size', self.__size),
                                                 ('limit', self.__limit_price),
                                                 ('stop', self.__stop_price),
                                                 ('sl', self.__sl_price),
                                                 ('tp', self.__tp_price),
                                                 ('contingent', self.is_contingent),
                                                 ('tag', self.__tag),
                                             ) if value is not None))
```

Which I haven't tested yet. Would just need an additional import from typing for SupportsRound.

Reply:
Something like that. Can also use [`isinstance(value, Number)`](https://docs.python.org/3/library/numbers.html#numbers.Number) like we do in [some other places](https://github.com/search?q=repo%3Akernc%2Fbacktesting.py+isinstance+number&type=code).

Reply:
```python
  def __repr__(self):
      return '<Order {}>'.format(', '.join(f'{param}={round(value, 5) if isinstance(value, (int, float)) else value}'
                                           for param, value in (
                                               ('size', self.__size),
                                               ('limit', self.__limit_price),
                                               ('stop', self.__stop_price),
                                               ('sl', self.__sl_price),
                                               ('tp', self.__tp_price),
                                               ('contingent', self.is_contingent),
                                               ('tag', self.__tag),
                                           ) if value is not None))
```